desdeo_emo/selection/APD_Select_constraints.py

```diff
@@ -83,6 +83,26 @@ def do(self, pop: Population, vectors: ReferenceVectors) -> List[int]:
             sub_population_index = np.atleast_1d(
                 np.squeeze(np.where(assigned_vectors == i))
             )
+
+            # Constraint check
+            if len(sub_population_index) > 1 and pop.constraint_violation is not None:
+                constraint_following_index = []
+                violation_values = pop.constraint_violation[sub_population_index]
+                for individual in sub_population_index:
+
+                    if((pop.constraint_violation[individual] < 0).all()):
+                        constraint_following_index.append(individual)
```

Comment:
You might not need this loop by the way. looping in python is inefficient. You can instead work entirely with np arrays (which will internally loop in C).

But this is fine